src/Pact/Compile.hs

```diff
@@ -149,13 +149,14 @@ doModule (EAtom n Nothing Nothing _:ESymbol k _:es) li ai =
     _ -> syntaxError ai "Empty module"
     where
       defOnly d = case d of
-        TDef {} -> return d
-        TNative {} -> return d
-        TConst {} -> return d
-        TSchema {} -> return d
-        TTable {} -> return d
-        TUse {} -> return d
-        TBless {} -> return d
+        TDef {}         -> return d
+        TNative {}      -> return d
+        TConst {}       -> return d
+        TSchema {}      -> return d
+        TTable {}       -> return d
+        TUse {}         -> return d
+        TBless {}       -> return d
+        TDefProperty {} -> return d
```

Comment:
Perhaps you could give an example of where / how properties would be defined.

Reply:
It's more that it begs the question, why are properties in meta at all.

A reason why is to indicate clearly what is code and what is model checking. Properties are  tightly integrated with code, more than unit tests, but the reason for a clear distinction is to make it obvious that the properties are not loaded on the blockchain, much like doctests might be if we were to add support for them in meta.

Also, it fits our architectural model thus far: model checking doesn't bleed out of `Pact/Analyze` -- we could CPP out model checking entirely from blockchain builds. 

I will give some thought to a design, but I first wanted to defend the whole reason why we're in `meta` in the first place. `defproperty` as a Term radically changes that, so I would argue that a change this momentous needs its own justification.

As a final note, I would still like to keep the idea of a `foo.property` file as a way to share property abstractions, but also to potentially move proofs out of blockchain code if desired. One thing I do think is strange is `use`ing a module to get only its properties -- you're putting that dep now on the blockchain, which if it's just for proofs is not very valuable.

Reply:
A simple design is just `def` or `defs` leading off a block, similarly to properties now. For importing properties from a file you could have `import`, or from another module, `use`. 
```lisp
(module foo 'ks
  (meta "foo docstring"
    (use bar "optional-hashvalue")
    (import "shared/stuff.properties")
    (defs [
      (def conserves-mass
        "Ensure a BALANCE column in ACCOUNTS table is net zero"
        (= (column-delta 'accounts 'balance) 0.0))
      (def auth-required
        "Always governed by accounts-admin-keyset"
        (authorized-by 'accounts-admin-keyset))
    ])
    ...
)
```

I'm trying out `defs` + `def` but it could also just be `(conserves-mass (= ...))`

Looking at that, it seems to me some level of argument/macro specification could be cool, ie:
```lisp
 (def conserves-mass (balance-col accounts-table)
        "Ensure a BALANCE-COL in ACCOUNTS-TABLE is net zero"
        (= (column-delta accounts-table balance-col) 0.0)) 
```

Reply:
Yeah Joel talks property args here: https://github.com/kadena-io/pact/issues/123#issue-333373446

Reply:
> One thing I do think is strange is `use`ing a module to get only its properties -- you're putting that dep now on the blockchain, which if it's just for proofs is not very valuable.

This makes sense to me. My first thought is to add syntax like `use properties` (rather than `import` to specify you're only importing properties). However, either way, you're mentioning a file which is not put on the blockchain, so it's basically creating a dead link. Can we avoid this?

I *think* defining properties within a `meta` would work. My only concerns are

1. does it feel cumbersome to use?
2. is it intuitive that the defined property is visible from the entire module / when the module is imported?
3. do we have to replicate a lot of the same machinery that already exists, but for the meta level?